### DISCUSSION TITLE
post/2020/2020-09-git-commit-message/

### DISCUSSION THREAD

**Discussion by giscus** [zyrikby/website_related#27] [Announcements]
# post/2020/2020-09-git-commit-message/

De facto, git is a standard version control system now. Not surprisingly that I use it for my projects. Working on a project, I often face with a situation when I have made a number of changes, and I need to describe them in a commit message. Of course, I try following the line when every change results in a new commit, however, on practice, sometime I face with the situation when I need to describe a number of changes in one commit message.  If there are a number of changes (or peculiarities), mentioning all of them could be a challenge. Of course, you can open an additional terminal and do diffing, however, it would be great if this information is included in the commit message.

https://zhauniarovich.com/post/2020/2020-09-git-commit-message/

**Comment by philipsd6** [comment]
Sorry to comment after two years, but this is an awful lot of work to do essentially:

`git diff --cached | sed 's/^/# /' >> $COMMIT_MSG_FILE`

But what's worse is this doesn't take into account whether or not the user has used the `-m` parameter to set their commit message on the cli, or if they're using `--amend` or squashing commits, etc. 

The good news is you don't need a hook for this anyway: just add `-v` to your commit call, or for a more permanent method, set it in your config with `git config --global commit.verbose 1`
